### DISCUSSION TITLE
File Provider API changes affect Dropbox. Same with Maestral?

### DISCUSSION THREAD

**Discussion by markelp** [samschott/maestral#836] [General]
Will these changes affect Maestral?

https://help.dropbox.com/installs/macos-support-for-expected-changes

**Comment by samschott** [comment]
No, lucky we are not affected. Maestral uses good old-fashioned [FSEvents](https://developer.apple.com/documentation/coreservices/file_system_events) to get notified of file changes.

Dropbox used to have its own kernel extension instead, which allowed them to react to local file changes more granularly compared to FSEvents. Kernel extensions are increasingly locked down on macOS for security and stability reasons, forcing Dropbox to switch. They decided to go with the [File Provider API](https://developer.apple.com/documentation/fileprovider) which is cross-platform for iOS and macOS and directly geared towards the "files in the cloud" use case. It does however come with a couple of trade-offs.
